### PR TITLE
Add thrift input format

### PR DESCRIPTION
**Pull request by bananaaggle**
Because of deprecated of parseSpec, I develop ThriftInputFormat for new interface, which supports stream ingestion for data encoded by Thrift.

This PR has:
- [ ] been self-reviewed.
   - [ ] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [ ] added documentation for new or modified features or behaviors.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/dev/license.md)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [ ] added integration tests.
- [ ] been tested in a test Druid cluster.

### CODE REVIEW

**Comment by bananaaggle** [comment]
Hi, @clintropolis! I think thrift is the last extension which use `parser`. When this `inputformat` finished, we can remove parser's implementations from code and fix all document about it. Do you think we should open an issue for it?

**Review by clintropolis — COMMENTED** [review]
very sorry for the delayed review, the changes overall lgtm :+1:

File: docs/ingestion/data-formats.md [hunk 1]
```diff
@@ -223,6 +223,41 @@ The Parquet `inputFormat` has the following components:
 |flattenSpec| JSON Object |Define a [`flattenSpec`](#flattenspec) to extract nested values from a Parquet file. Note that only 'path' expression are supported ('jq' is unavailable).| no (default will auto-discover 'root' level properties) |
 | binaryAsString | Boolean | Specifies if the bytes parquet column which is not logically marked as a string or enum type should be treated as a UTF-8 encoded string. | no (default = false) |
 
+### Thrift Stream
```
[the remark this answers is not in the document]
Comment (techdocsmith) [reply docs/ingestion/data-formats.md:226]: +1 for @clintropolis suggestion to keep the doc in /docs/development/extensions-contrib/thrift.md until the extension is made core.

**Comment by abhishekagarwal87** [comment]
@bananaaggle - can you resolve these conflicts? it will be great to get rid of the dependency on `InputRowParser` completely.

**Pull request closed without merging**